README.md

```diff
@@ -1,33 +1,355 @@
+img.mosaic_tile_dims.Y  # 512 (for example)
+
+# Get the tile start indices (top left corner of tile)
+y_start_index, x_start_index = img.get_mosaic_tile_position(12)
+```
+
+### Metadata Reading
+
+```python
+from bioio import BioImage
+
+# Get an BioImage object
+img = BioImage("my_file.tiff")  # selects the first scene found
+img.metadata  # returns the metadata object for this file format (XML, JSON, etc.)
+img.channel_names  # returns a list of string channel names found in the metadata
+img.physical_pixel_sizes.Z  # returns the Z dimension pixel size as found in the metadata
+img.physical_pixel_sizes.Y  # returns the Y dimension pixel size as found in the metadata
+img.physical_pixel_sizes.X  # returns the X dimension pixel size as found in the metadata
+```
+
+### Xarray Coordinate Plane Attachment
+
+If `bioio` finds coordinate information for the spatial-temporal dimensions of
+the image in metadata, you can use
+[xarray](http://xarray.pydata.org/en/stable/index.html) for indexing by coordinates.
+
+```python
+from bioio import BioImage
+
+# Get an BioImage object
+img = BioImage("my_file.ome.tiff")
+
+# Get the first ten seconds (not frames)
+first_ten_seconds = img.xarray_data.loc[:10]  # returns an xarray.DataArray
+
+# Get the first ten major units (usually micrometers, not indices) in Z
+first_ten_mm_in_z = img.xarray_data.loc[:, :, :10]
+
+# Get the first ten major units (usually micrometers, not indices) in Y
+first_ten_mm_in_y = img.xarray_data.loc[:, :, :, :10]
+
+# Get the first ten major units (usually micrometers, not indices) in X
+first_ten_mm_in_x = img.xarray_data.loc[:, :, :, :, :10]
+```
+
+See `xarray`
+["Indexing and Selecting Data" Documentation](http://xarray.pydata.org/en/stable/indexing.html)
+for more information.
+
+### Cloud IO Support
+
+[File-System Specification (fsspec)](https://github.com/intake/filesystem_spec) allows
+for common object storage services (S3, GCS, etc.) to act like normal filesystems by
+following the same base specification across them all. BioIO utilizes this
+standard specification to make it possible to read directly from remote resources when
+the specification is installed.
+
+```python
+from bioio import BioImage
+
+# Get an BioImage object
+img = BioImage("http://my-website.com/my_file.tiff")
+img = BioImage("s3://my-bucket/my_file.tiff")
+img = BioImage("gcs://my-bucket/my_file.tiff")
+
+# Or read with specific filesystem creation arguments
+img = BioImage("s3://my-bucket/my_file.tiff", fs_kwargs=dict(anon=True))
+img = BioImage("gcs://my-bucket/my_file.tiff", fs_kwargs=dict(anon=True))
+
+# All other normal operations work just fine
+```
+
+Remote reading requires that the file-system specification implementation for the
+target backend is installed.
+
+-   For `s3`: `pip install s3fs`
+-   For `gs`: `pip install gcsfs`
+
+See the [list of known implementations](https://filesystem-spec.readthedocs.io/en/latest/?badge=latest#implementations).
+
+### Saving to OME-TIFF
```

Comment:
We should add a section about writing to ome-zarr.  This also impacts the cloud support section above because ome-zarr supports writing direct to s3.

Reply:
Honestly I believe there is a lot of room for improvement with ome-zarr writing so we could shelf this as a future To-do item.